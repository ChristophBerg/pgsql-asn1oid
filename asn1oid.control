default_version = '1'
comment = 'asn1oid extension'
relocatable = true
