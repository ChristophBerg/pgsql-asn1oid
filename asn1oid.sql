CREATE TYPE asn1oid;
CREATE FUNCTION asn1oid_input (cstring)
    RETURNS asn1oid
    IMMUTABLE
    STRICT
    LANGUAGE 'C'
    AS 'asn1oid.so','asn1oid_input';
CREATE FUNCTION asn1oid_output (asn1oid)
    RETURNS cstring
    IMMUTABLE
    STRICT
    LANGUAGE 'C'
    AS 'asn1oid.so','asn1oid_output';
CREATE TYPE asn1oid (
    INPUT = asn1oid_input,
    OUTPUT = asn1oid_output);
CREATE FUNCTION asn1oid_eq (asn1oid, asn1oid)
    RETURNS bool
    IMMUTABLE
    STRICT
    LANGUAGE 'C'
    AS 'asn1oid.so','asn1oid_eq';
CREATE FUNCTION asn1oid_ne (asn1oid, asn1oid)
    RETURNS bool
    IMMUTABLE
    STRICT
    LANGUAGE 'C'
    AS 'asn1oid.so','asn1oid_ne';
CREATE FUNCTION asn1oid_lt (asn1oid, asn1oid)
    RETURNS bool
    IMMUTABLE
    STRICT
    LANGUAGE 'C'
    AS 'asn1oid.so','asn1oid_lt';
CREATE FUNCTION asn1oid_gt (asn1oid, asn1oid)
    RETURNS bool
    IMMUTABLE
    STRICT
    LANGUAGE 'C'
    AS 'asn1oid.so','asn1oid_gt';
CREATE FUNCTION asn1oid_le (asn1oid, asn1oid)
    RETURNS bool
    IMMUTABLE
    STRICT
    LANGUAGE 'C'
    AS 'asn1oid.so','asn1oid_le';
CREATE FUNCTION asn1oid_ge (asn1oid, asn1oid)
    RETURNS bool
    IMMUTABLE
    STRICT
    LANGUAGE 'C'
    AS 'asn1oid.so','asn1oid_ge';
CREATE FUNCTION asn1oid_cmp (asn1oid, asn1oid)
    RETURNS int4
    IMMUTABLE
    STRICT
    AS 'asn1oid.so','asn1oid_cmp' LANGUAGE 'C';
CREATE OPERATOR = (
    LEFTARG = asn1oid,
    RIGHTARG = asn1oid,
    PROCEDURE = asn1oid_eq,
    COMMUTATOR = '=',
    NEGATOR = '<>',
    RESTRICT = eqsel,
    JOIN = eqjoinsel,
    HASHES,
    MERGES);
CREATE OPERATOR < (
    LEFTARG = asn1oid,
    RIGHTARG = asn1oid,
    PROCEDURE = asn1oid_lt,
    COMMUTATOR = '>',
    NEGATOR = '>=',
    RESTRICT = scalarltsel,
    JOIN = scalarltjoinsel);
CREATE OPERATOR > (
    LEFTARG = asn1oid,
    RIGHTARG = asn1oid,
    PROCEDURE = asn1oid_gt,
    COMMUTATOR = '<',
    NEGATOR = '<=',
    RESTRICT = scalargtsel,
    JOIN = scalargtjoinsel);
CREATE OPERATOR <= (
    LEFTARG = asn1oid,
    RIGHTARG = asn1oid,
    PROCEDURE = asn1oid_le,
    COMMUTATOR = '>=',
    NEGATOR = '>',
    RESTRICT = scalarltsel,
    JOIN = scalarltjoinsel);
CREATE OPERATOR >= (
    LEFTARG = asn1oid,
    RIGHTARG = asn1oid,
    PROCEDURE = asn1oid_ge,
    COMMUTATOR = '<=',
    NEGATOR = '<',
    RESTRICT = scalargtsel,
    JOIN = scalargtjoinsel);
CREATE OPERATOR <> (
    LEFTARG = asn1oid,
    RIGHTARG = asn1oid,
    PROCEDURE = asn1oid_ne,
    COMMUTATOR = '<>',
    NEGATOR = '=',
    RESTRICT = neqsel,
    JOIN = neqjoinsel);
CREATE OPERATOR CLASS asn1oid_ops
    DEFAULT FOR TYPE asn1oid
    USING btree AS
        OPERATOR 1 <,
        OPERATOR 2 <=,
        OPERATOR 3 =,
        OPERATOR 4 >=,
        OPERATOR 5 >,
        FUNCTION 1 asn1oid_cmp(asn1oid, asn1oid);
