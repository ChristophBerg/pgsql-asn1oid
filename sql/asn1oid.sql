SELECT '1.2.3'::asn1oid;
SELECT '.1.2.3'::asn1oid;
SELECT null::asn1oid;
SELECT 'foo'::asn1oid;
SELECT 123::asn1oid;
