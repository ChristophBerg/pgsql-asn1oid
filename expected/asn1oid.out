SELECT '1.2.3'::asn1oid;
 asn1oid 
---------
 1.2.3
(1 row)

SELECT '.1.2.3'::asn1oid;
 asn1oid 
---------
 1.2.3
(1 row)

SELECT null::asn1oid;
 asn1oid 
---------
 
(1 row)

SELECT 'foo'::asn1oid;
ERROR:  invalid input syntax for type asn1oid: "foo"
LINE 1: SELECT 'foo'::asn1oid;
               ^
SELECT 123::asn1oid;
ERROR:  cannot cast type integer to asn1oid
LINE 1: SELECT 123::asn1oid;
                  ^
